app/helpers/check_answers_helper.rb

```diff
@@ -1,17 +1,31 @@
 module CheckAnswersHelper
-  def check_answer_link(url:, question:, answer:, name:, is_long: nil)
-    partial = case answer
-              when Hash
-                'currency_items'
-              when Array
-                'items'
-              else
-                'item'
-              end
-    is_long = '--long' if is_long
+  def check_answer_link(url:, question:, answer:, name:)
     render(
-      partial: "shared/check_answers/#{partial}",
-      locals: { url: url, question: question, answer: answer, name: name, is_long: is_long }
+      'shared/check_answers/item',
+      name: name,
+      url: url,
+      question: question,
+      answer: answer
+    )
+  end
```

Comment:
The main reason I split the group methods below, out of `check_answer_link` was because they need to be used differently.  `check_answer_link` provides one items within a list - and therefore needs enclosing `dl` tags. So:
```erb
<dl class="govuk-summary-list govuk-!-margin-bottom-9">
  <%= check_answer_link ..... %>
</dl>
```

The two list methods below create both the items and the `dl` that contains them, so don't need to be enclosed within a `dl` tag